### DISCUSSION TITLE
Enable automatic escaping of all stuff rendered through ".html.jinja" templates

### DISCUSSION THREAD

**Discussion by DerMax93** [pallets/quart#318] [Q&A] [answered]
Took me a while to figure out why quart suddenly stopped escaping the variables i rendered. 
As it turns out if you rename your templates to "whatever.html.jinja" (was a suggestion to make vscode shut up about the jinja code in them) quart/jinja (idk where it happens) stops escaping the strings you pass into the templates. The rest continues to work as expected. You can even re-enable escaping in the templates on a per-value basis (with the "|e" pipe thingy). For now I have renamed back to "just.html", but I am wondering whether there is a way to get it to escape both types...

**Comment by davidism** [comment, marked as answer]
You can override `Quart.select_jinja_autoescape(filename)` to return `True` when a given filename should have autoescape enabled. The default looks for the extensions `html`, `htm`, `xml`, `xhtml`, and `svg`.

I'm also pretty sure there's a few extensions for VS Code that add Jinja template support.

**Reply by DerMax93** [reply]
Thank you very much :D
